File: app/src/main/java/com/wolcano/musicplayer/music/mvp/listener/TaskInterface.java
package com.wolcano.musicplayer.music.mvp.listener;


import com.wolcano.musicplayer.music.mvp.models.Song;

import java.util.List;


public interface TaskInterface<T> {
    void onPrepare();
    void onTaskDone(List<Song> songList);
    void onTask();
    void onTaskFail(Exception e);
}

File: app/src/main/java/com/wolcano/musicplayer/music/mvp/listener/TaskInterface.kt
package com.wolcano.musicplayer.music.mvp.listener

import com.wolcano.musicplayer.music.model.Song

interface TaskInterface<T> {
    fun onPrepare()
    fun onTaskDone(songList: List<Song>?)
    fun onTask()
    fun onTaskFail(e: Exception?)
}